packages/js-dash-sdk/src/SDK/Client/Platform/methods/contracts/update.ts

```diff
@@ -25,13 +26,33 @@ export default async function update(
 
   updatedDataContract.incrementVersion();
 
+  const identityId = identity.getId();
+  const dataContractId = dataContract.getId();
+
+  const identityContractNonce = await this.nonceManager
+    .getIdentityContractNonce(identityId, dataContractId) + 1;
+
   const dataContractUpdateTransition = dpp.dataContract
-    .createDataContractUpdateTransition(updatedDataContract);
+    .createDataContractUpdateTransition(updatedDataContract, BigInt(identityContractNonce));
 
   this.logger.silly(`[DataContract#update] Created data contract update transition ${dataContract.getId()}`);
 
   await signStateTransition(this, dataContractUpdateTransition, identity, 2);
-  await broadcastStateTransition(this, dataContractUpdateTransition);
+  try {
+    // Broadcast state transition also wait for the result to be obtained
+    await broadcastStateTransition(this, dataContractUpdateTransition);
+    this.nonceManager
+      .setIdentityContractNonce(identityId, dataContractId, identityContractNonce);
+  } catch (e) {
+    // Deadline exceeded would mean that state transition didn't make it to the block,
+    // so we will not update nonce in this case
+    if (e.code !== GrpcErrorCodes.DEADLINE_EXCEEDED) {
```

Comment:
It doesn't guarantee that nonce is actually incremented. It's updated only in case if it was a paid consensus error.